test/hotspot/jtreg/serviceability/jvmti/vthread/VirtualThreadStartTest/libVirtualThreadStartTest.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <cstdlib>
+#include <cstring>
+#include <jvmti.h>
+#include "jvmti_common.h"
+
+extern "C" {
+
+static jvmtiEnv *jvmti;
+static int started_thread_cnt = 0;
+static jrawMonitorID agent_event_lock = NULL;
+static const char* TESTED_TNAME_START = "Tested-VT";
+static const size_t TESTED_TNAME_START_LEN = strlen(TESTED_TNAME_START);
+static bool can_support_vt_enabled = false;
+
+void JNICALL ThreadStart(jvmtiEnv *jvmti, JNIEnv* jni, jthread thread) {
+  char* tname = get_thread_name(jvmti, jni, thread);
+
+  RawMonitorLocker agent_start_locker(jvmti, jni, agent_event_lock);
+
+  if (tname != NULL && strncmp(tname, TESTED_TNAME_START, TESTED_TNAME_START_LEN) == 0) {
+    jboolean is_virtual = jni->IsVirtualThread(thread);
+    if (!is_virtual) {
+      fatal(jni, "Failed: tested thread expected to be virtual");
+    }
+    if (can_support_vt_enabled) {
+      fatal(jni, "Failed: expected VirtualThreadStart instead of ThreadStart event");
+    }
+    printf("ThreadStart event: %s\n", tname);
+    started_thread_cnt++;
+  }
+  deallocate(jvmti, jni, (void*)tname);
```

Comment:
This will crash if `get_thread_name()` returns the string constant `"<Unnamed thread>"`

Reply:
Nice catch.
It is strange I've never saw any related crashes.
This could be addressed separately but I've fixed it now.
Please, let me know if you are okay with it.